PhysicsTools/NanoAOD/plugins/VertexTableProducer.cc

```diff
@@ -93,6 +101,35 @@ VertexTableProducer::VertexTableProducer(const edm::ParameterSet& params)
       dlenSigMin_(params.getParameter<double>("dlenSigMin"))
 
 {
+  if (params.existsAs<edm::ParameterSet>("optionalPvVariables")) {
```

Comment:
please remove the `optionalPvVariables` (and DumpedVariables.h) that you are not using, and that can be implemented using `SimpleVertexFlatTableProducer` as you did, and should be done

Reply:
The issue with using `SimpleVertexFlatTableProducer` in this context is that it is not possible to simply extend the "main" PV table produced by `VertexTableProducer.cc`. One can add a new table, even with the same name as the main PV one, but extension does not work. It is a scheduling problem. Maybe it is also because the main PV table is declared as singleton, while to produce table with 1 item from multi-item collection one should do the trick by setting max no. of inputs to 1 (it takes the leading input which is fine for PV). 
When the two PV tables (main and additional) have same name it is not visible a final user but although it works with NANOAOD output module I am not sure if it will work with NANOAODEDM output module which I have not tried.
Maybe I overlooked something - do you have any suggestions? And what is a problem with current implementation? It is too specific?

Reply:
Any further comments, suggestions?

Reply:
please follow the previous suggestion. if there's an issue with the extension, please fill a GH issue, or in the XPOG gitlab and we'll figure this out

Reply:
OK.